rs/bitcoin/adapter/src/router.rs

```diff
@@ -46,16 +46,15 @@ pub fn start_main_event_loop(
         network_message_sender,
         router_metrics.clone(),
     );
+    let adapter_awake_notifier = adapter_state.awake_notifier();
 
     tokio::task::spawn(async move {
         let mut tick_interval = interval(Duration::from_millis(100));
         loop {
-            let sleep_idle_interval = Duration::from_millis(100);
             if adapter_state.is_idle() {
                 connection_manager.make_idle();
                 blockchain_manager.make_idle();
-                // TODO: instead of sleeping here add some async synchronization.
-                sleep(sleep_idle_interval).await;
+                adapter_awake_notifier.notified().await;
                 continue;
```

Comment:
I think you can remove the continue here

Reply:
done